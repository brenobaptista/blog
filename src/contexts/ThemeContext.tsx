import { createContext } from 'react'
import { DefaultTheme } from 'styled-components'

import dracula from '@/styles/themes/dracula'

interface IThemeContext {
  theme: DefaultTheme
  toggleTheme: () => void
}

const ThemeContext = createContext<IThemeContext>({
  theme: dracula,
  toggleTheme: () => null
})

export default ThemeContext
