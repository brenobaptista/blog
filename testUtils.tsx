import { render } from '@testing-library/react'
import { RenderResult } from '@testing-library/react/types'
import { ThemeProvider } from 'styled-components'

import light from './src/styles/themes/light'

const Providers = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
)

const customRender = (ui: React.ReactElement, options = {}): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
