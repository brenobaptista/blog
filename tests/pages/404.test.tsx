import { render } from 'testUtils'
import Custom404 from '@/pages/404'
import ThemeContext from '@/contexts/ThemeContext'

const renderWithContext = (ui, value) =>
  render(<ThemeContext.Provider value={value}>{ui}</ThemeContext.Provider>)

test('matches snapshot in dracula mode', () => {
  const theme = { mode: 'dracula' }
  const { asFragment } = renderWithContext(<Custom404 />, {
    theme
  })

  expect(asFragment()).toMatchSnapshot()
})
