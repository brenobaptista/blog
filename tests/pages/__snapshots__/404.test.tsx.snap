// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot in dracula mode 1`] = `
<DocumentFragment>
  .c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  font-size: 0.9rem;
}

.c9 a {
  margin: 16px;
  border-radius: 4px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  border: 0;
  height: 1px;
  background-color: gray;
  background-image: linear-gradient(to right,gray,gray,gray);
}

.c3 {
  position: absolute;
  height: 0;
  width: 0;
  visibility: hidden;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  cursor: url('/images/retro/cursor/pointer.cur'),pointer;
  border-radius: 56px;
  width: 56px;
  height: 28px;
  background-color: orange;
}

.c5:hover {
  background-color: darkorange;
}

.c7 {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  border-radius: 24px;
  width: 24px;
  height: 24px;
  background-color: black;
  -webkit-transition: 0.2s all ease-in-out;
  transition: 0.2s all ease-in-out;
}

.c2:checked + .c4 .c6 {
  left: calc(100% - 2px);
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c0 {
  max-width: 768px;
  padding: 16px;
  margin: 32px auto;
  background-color: black;
  border: 1px solid gray;
  border-radius: 8px;
  font-family: Fira Code,monospace;
}

.c0 img {
  max-width: 100%;
  display: block;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 32px;
}

.c1 a {
  position: relative;
  top: -8px;
  font-size: 1.4rem;
  font-weight: bold;
}

@media screen and (max-width:360px) {
  .c9 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    text-align: center;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    margin: 16px 8px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        href="/"
      >
        Breno Baptista
      </a>
      <input
        checked=""
        class="c2 c3"
        id="checkbox"
        type="checkbox"
      />
      <label
        aria-label="Change the visual styles"
        class="c4 c5"
        data-testid="label-checkbox"
        for="checkbox"
        tabindex="0"
      >
        <span
          class="c6 c7"
        />
      </label>
    </div>
    <main>
      <h1>
        404: NOT FOUND
      </h1>
      <p>
        It seems you have found something that used to exist or you spelled something wrong. But at least you have found Pitucho:
      </p>
      <img
        alt="Pitucho"
        src="/images/pitucho.jpg"
      />
    </main>
    <footer>
      <hr
        class="c8"
      />
      <nav
        class="c9"
      >
        <div
          class="c10"
        >
          <a
            href="mailto:brenobaptista@protonmail.com"
          >
            Email
          </a>
        </div>
        <div
          class="c10"
        >
          <a
            href="https://github.com/brenobaptista/"
            rel="noreferrer noopener"
            target="_blank"
          >
            GitHub
          </a>
        </div>
        <div
          class="c10"
        >
          <a
            href="/rss.xml"
          >
            RSS
          </a>
        </div>
        <div
          class="c10"
        >
          <a
            href="/resume"
          >
            Résumé
          </a>
        </div>
      </nav>
    </footer>
  </div>
</DocumentFragment>
`;
